frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -60,55 +63,56 @@ export const SettingsPage = (): JSX.Element => {
       navigate(`${ADMINFORM_ROUTE}/${formId}/${ADMINFORM_RESULTS_SUBROUTE}`)
   }, [formId, hasEditAccess, isCollabLoading, navigate])
 
-  const tabConfig = useMemo(() => {
-    const emailsNotificationsTab =
-      settings?.responseMode === FormResponseMode.Encrypt ||
-      settings?.responseMode === FormResponseMode.Email
-        ? {
-            label: 'Email notifications',
-            icon: BiMailSend,
-            component: SettingsEmailsPage,
-            path: 'email-notifications',
-            showRedDot: true,
-          }
-        : null
-
-    const baseConfig: (TabEntry | null)[] = [
-      {
-        label: 'General',
-        icon: BiCog,
-        component: SettingsGeneralPage,
-        path: 'general',
-      },
-      {
-        label: 'Singpass',
-        icon: BiKey,
-        component: SettingsAuthPage,
-        path: 'singpass',
-      },
-      emailsNotificationsTab,
-      {
-        label: 'Twilio credentials',
-        icon: BiMessage,
-        component: SettingsTwilioPage,
-        path: 'twilio-credentials',
-      },
-      {
-        label: 'Webhooks',
-        icon: BiCodeBlock,
-        component: SettingsWebhooksPage,
-        path: 'webhooks',
-      },
-      {
-        label: 'Payments',
-        icon: BiDollar,
-        component: SettingsPaymentsPage,
-        path: 'payments',
-      },
-    ]
-
-    return baseConfig.filter(isNonEmpty)
-  }, [settings?.responseMode])
+  const isTest = process.env.NODE_ENV === 'test'
+  // For beta flagging email notifications tab.
+  const emailNotificationsTab =
+    isUserLoading ||
+    isFormSettingLoading ||
+    (!isTest &&
+      formSettings?.responseMode === FormResponseMode.Multirespondent &&
+      !user?.betaFlags?.mrfEmailNotifications)
+      ? null
+      : {
+          label: 'Email notifications',
+          icon: BiMailSend,
+          component: SettingsEmailsPage,
+          path: 'email-notifications',
+          showRedDot: true,
+        }
+
+  const tabConfig: TabEntry[] = [
+    {
+      label: 'General',
+      icon: BiCog,
+      component: SettingsGeneralPage,
+      path: 'general',
+    },
+    {
+      label: 'Singpass',
+      icon: BiKey,
+      component: SettingsAuthPage,
+      path: 'singpass',
+    },
+    emailNotificationsTab,
+    {
+      label: 'Twilio credentials',
+      icon: BiMessage,
+      component: SettingsTwilioPage,
+      path: 'twilio-credentials',
+    },
+    {
+      label: 'Webhooks',
+      icon: BiCodeBlock,
+      component: SettingsWebhooksPage,
+      path: 'webhooks',
+    },
+    {
+      label: 'Payments',
+      icon: BiDollar,
+      component: SettingsPaymentsPage,
+      path: 'payments',
+    },
+  ].filter((tab) => tab !== null) as TabEntry[]
```

Comment:
/nit

```suggestion
  ].filter(Boolean) as TabEntry[]
```

Reply:
Thanks for this suggestion. Learnt something new about JS today.